unit-tests/internal/internal-tests-extrinsic.cpp

```diff
@@ -26,8 +36,90 @@ inline void require_identity_matrix(const float(&matrix)[9])
 {
     static const float identity_matrix_3x3[] = { 1,0,0, 0,1,0, 0,0,1 };
     for (int i = 0; i < 9; ++i) REQUIRE(matrix[i] == approx(identity_matrix_3x3[i]));
+}*/
+
+bool get_mode(rs2::device& dev, rs2::stream_profile& profile, int mode_index = 0)
+{
+    auto sensors = dev.query_sensors();
+    REQUIRE(sensors.size() > 0);
+
+    for (auto i = 0; i < sensors.size(); i++)
+    {
+        auto modes = sensors[i].get_stream_profiles();
+        REQUIRE(modes.size() > 0);
+
+        if (mode_index >= modes.size())
+            continue;
+
+        profile = modes[mode_index];
+        return true;
+    }
+    return false;
+}
+
+double line_fitting(const std::vector<double>& y_vec, std::vector<double>& y_fit)
```

Comment:
Add function documentation with in/out parameters description.
also specify the external reference for the algo,
You also need to extract the C value, not only slope for  `y = a*x + C`